### PR TITLE
feat(autoscaling): add new check `autoscaling_group_elb_health_check_enabled`

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

This new check ensures that Amazon EC2 Auto Scaling groups associated with a load balancer use Elastic Load Balancing (ELB) health checks. ELB health checks help verify the health of instances based on additional tests provided by the load balancer, thereby supporting the availability and reliability of applications using EC2 Auto Scaling groups.

### Description

Added new check `autoscaling_group_elb_health_check_enabled` with its unit tests.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/autoscaling/autoscaling_group_health_check_elb_enabled/autoscaling_group_health_check_elb_enabled.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/autoscaling/autoscaling_group_health_check_elb_enabled/autoscaling_group_health_check_elb_enabled.py:1]: This check only applies to autoscaling groups that are associated to Elastic Load Balancers. If not, no findings should be created.

Comment (MrCloudSec) [reply prowler/providers/aws/services/autoscaling/autoscaling_group_health_check_elb_enabled/autoscaling_group_health_check_elb_enabled.py:1]: Please, show in the status extended which check type has if it is not ELB.

Comment (danibarranqueroo) [reply prowler/providers/aws/services/autoscaling/autoscaling_group_health_check_elb_enabled/autoscaling_group_health_check_elb_enabled.py:1]: Done, I've verify if the describe-auto-scaling-groups command output returns an empty array (i.e. []) for both "LoadBalancers" and "TargetGroups", meaning this that there are no ELB associated.

Comment (MrCloudSec) [reply prowler/providers/aws/services/autoscaling/autoscaling_group_health_check_elb_enabled/autoscaling_group_health_check_elb_enabled.py:1]: Awesome, thanks!

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**